scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamMode.scala

```diff
@@ -96,6 +101,27 @@ class BeamFileIO[T](output: String) extends BeamSink[T] {
   }
 }
 
+class BeamTempFileSource[T](coder: Coder[T], output: String) extends BeamSource[T] {
+  override def read(
+      pipeline: Pipeline,
+      config: Config
+  ): PCollection[_ <: T] =
+    pipeline
+      .apply(Create.of(s"$output/*"))
+      .apply(FileIO.matchAll())
+      .apply(ParDo.of(new TempSourceDoFn[T](coder)))
+      .setCoder(coder)
+}
+
+case class TempSourceDoFn[T](coder: Coder[T]) extends DoFn[MatchResult.Metadata, T] {
+  @ProcessElement
+  def processElement(c: DoFn[MatchResult.Metadata, T]#ProcessContext): Unit = {
```

Comment:
can you add a comment to the code here? It looks to me like everything in the `Iterator` is going to be read in a single thread.

I can't really see how that's not the case? That seems like it is going to really hard performance since you don't want to kill parallelism when using forceToDiskExecution.